File: src/main/java/com/example/hnefatafl/exception/GlobalExceptionHandler.java
package com.example.hnefatafl.exception;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@ControllerAdvice
public class GlobalExceptionHandler {
    @ExceptionHandler(ObjectNotFoundException.class)
    public ResponseEntity<String> handleGameNotFound(ObjectNotFoundException ex) {
        return ResponseEntity.status(404).body(ex.getMessage());
    }

    @ExceptionHandler(ObjectAlreadyExistsException.class)
    public ResponseEntity<String> handleObjectAlreadyExists(ObjectAlreadyExistsException ex) {
        return ResponseEntity.status(404).body(ex.getMessage());
    }

    @ExceptionHandler(IndexOutOfBoundsException.class)
    public ResponseEntity<String> handleIndexOutOfBound(IndexOutOfBoundsException ex) {
        String identify = "";
        for (StackTraceElement stack:ex.getStackTrace()) {
            if( stack.getClassLoaderName()=="app") {
                identify = stack.getFileName() + "[" + stack.getLineNumber() + "] - ";
                break;
            }
        }

        return ResponseEntity.status(404).body(identify + ex.getMessage());
    }
}

File: src/main/java/com/example/hnefatafl/exception/GlobalExceptionHandler.kt
package com.example.hnefatafl.exception

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler

@ControllerAdvice
class GlobalExceptionHandler {
    @ExceptionHandler(ObjectNotFoundException::class)
    fun handleGameNotFound(ex: ObjectNotFoundException): ResponseEntity<String?> {
        return ResponseEntity.status(404).body(ex.message)
    }

    @ExceptionHandler(ObjectAlreadyExistsException::class)
    fun handleObjectAlreadyExists(ex: ObjectAlreadyExistsException): ResponseEntity<String?> {
        return ResponseEntity.status(404).body(ex.message)
    }

    @ExceptionHandler(IndexOutOfBoundsException::class)
    fun handleIndexOutOfBound(ex: IndexOutOfBoundsException): ResponseEntity<String> {
        var identify = ""
        for (stack in ex.stackTrace) {
            if (stack.classLoaderName === "app") {
                identify = stack.fileName + "[" + stack.lineNumber + "] - "
                break
            }
        }
        return ResponseEntity.status(404).body(identify + ex.message)
    }
}